ruma-identity-service-api/src/association/email/create_email_validation_session/v2.rs

```diff
@@ -0,0 +1,56 @@
+//! [POST /_matrix/identity/v2/validate/email/requestToken](https://matrix.org/docs/spec/identity_service/r0.3.0#post-matrix-identity-v2-validate-email-requesttoken)
+
+use ruma_api::ruma_api;
+
+ruma_api! {
+    metadata: {
+        description: "Create a session for validating an email address.",
+        method: POST,
+        name: "create_email_validation_session",
+        path: "/_matrix/identity/v2/validate/email/requestToken",
+        authentication: AccessToken,
+        rate_limited: false,
+    }
+
+    request: {
+        /// A unique string generated by the client, and used to identify the validation attempt.
+        pub client_secret: &'a str,
+
+        /// The email address to validate.
+        pub email: &'a str,
+
+        /// The server will only send an email if the send_attempt is a number greater than the
+        /// most recent one which it has seen, scoped to that email + client_secret pair.
+        pub send_attempt: js_int::UInt,
+
+        /// When the validation is completed, the identity server will redirect the user to this
+        /// URL.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub next_link: Option<&'a str>,
+    }
+
+    response: {
+        /// The session ID. Session IDs are opaque strings generated by the identity server.
+        pub sid: String,
+    }
+}
+
+impl<'a> Request<'a> {
+    /// Create a new `Request` with the given client secret, email ID, and the `send_attempt`
+    /// number.
```

Comment:
I don't have an opinion on whether `next_link` should be part of the argument list (though I guess it's "safer" in a way to include it), but the docs should match either way.

Reply:
It should also be `Creates` instead of `Create`,  again for consistency.

Reply:
@abhikjain360, let's keep `next_link` as a parameter.